packages/tc-schema-validator/tests/type-test-suite.js

```diff
@@ -420,50 +422,54 @@ const relationshipTestSuite = type => {
 			properties,
 		});
 
-		describe('from', () => {
-			it('has existing from type', () => {
+		describe('relationship endpoints', () => {
+			it('uses existing record type fro from', () => {
 				const fromType = types.find(t => t.name === from.type);
 				expect(fromType).toBeDefined();
 				// assert the relationship node doesn't have more relationship
 				expect(isRichRelationshipType(fromType)).toBe(false);
 			});
-			it('relationship must be closed', () => {
-				const endType = types.find(t => t.name === to.type);
-				const propExists = Object.values(endType.properties).filter(
-					prop => prop.type === type.name,
-				);
-				expect(propExists.length).toBeGreaterThan(0);
-				// If from and to type are the same, we should validate direction
-				if (from.type === to.type) {
-					const closeRel = propExists.find(
-						prop => prop.direction === 'to',
-					);
-					expect(closeRel).toBeDefined();
-				}
-			});
-		});
 
-		describe('to', () => {
-			it('has existing to type', () => {
+			it('uses existing record type for to', () => {
 				const toType = types.find(t => t.name === to.type);
 				expect(toType).toBeDefined();
 				// assert the relationship node doesn't have more relationship
 				expect(isRichRelationshipType(toType)).toBe(false);
 			});
-			it('relationship must be closed', () => {
-				const endType = types.find(t => t.name === from.type);
-				const propExists = Object.values(endType.properties).filter(
-					prop => prop.type === type.name,
-				);
-				expect(propExists.length).toBeGreaterThan(0);
-				// If from and to type are the same, we should validate direction
-				if (from.type === to.type) {
-					const closeRel = propExists.find(
-						prop => prop.direction === 'from',
+
+			if (from.type !== to.type) {
+				it('from type makes use of this relationship type', () => {
+					const endType = types.find(t => t.name === to.type);
+					const propertiesUsingRelationshipType = Object.values(
+						endType.properties,
+					).filter(prop => prop.type === type.name);
+					expect(propertiesUsingRelationshipType.length).toBe(1);
+				});
+				it('to type makes use of this relationship type', () => {
+					const endType = types.find(t => t.name === from.type);
+					const propertiesUsingRelationshipType = Object.values(
+						endType.properties,
```

Comment:
didn't get this assertion - for example in `CuriousChild` I don't see where `prop.type === type.name` becomes true. am I missing something